src/compile/scale/component.ts

```diff
@@ -31,16 +31,37 @@ export class ScaleComponent extends Split<ScaleComponentProps> {
   }
 
   /**
-   * Whether the scale definitely includes zero in the domain
+   * Whether the scale definitely includes or not include zero in the domain
    */
-  public domainDefinitelyIncludesZero() {
-    if (this.get('zero') !== false) {
-      return true;
+  public domainHasZero(): 'definitely' | 'definitely-not' | 'maybe' {
+    const scaleType = this.get('type');
+    if (contains([ScaleType.LOG, ScaleType.TIME, ScaleType.UTC], scaleType)) {
+      // Log scales cannot have zero.
+      // Zero in time scale is arbitrary, and does not affect ratio.
+      // (Time is an interval level of measurement, not ratio).
+      // See https://en.wikipedia.org/wiki/Level_of_measurement for more info.
+      return 'definitely-not';
     }
-    return some(
-      this.get('domains'),
-      d => isArray(d) && d.length === 2 && isNumber(d[0]) && d[0] <= 0 && isNumber(d[1]) && d[1] >= 0
-    );
+
+    const scaleZero = this.get('zero');
+    if (
+      scaleZero === true ||
```

Comment:
what's the type of `scaleZero`? boolean | undefined? If so, I might just use `scaleZero` to avoid checking with boolean.

Reply:
yes we can use `scaleZero` too, but I think this reads clearer.